lib/Sema/ConstraintSystem.cpp

```diff
@@ -1175,7 +1175,8 @@ ConstraintSystem::getTypeOfMemberReference(
          func->hasArchetypeSelf())) {
       openedType = openedType->replaceCovariantResultType(
                      baseObjTy,
-                     func->getNumParameterLists());
+                     func->getNumParameterLists(),
+                     !baseObjTy->getAnyOptionalObjectType());
```

Comment:
@DougGregor Thanks! Actually if you look at callers of `replaceCovariantResultType` there are none which actually use it, so do you think it would be good idea to simply remove `preserveOptionality` all together as well?